application/src/main/java/org/togetherjava/tjbot/features/javamail/RSSHandlerRoutine.java

```diff
@@ -0,0 +1,412 @@
+        Optional<ZonedDateTime> lastSavedDate = getLastSavedDateFromDatabaseRecord(rssFeedRecord,
+                feedConfig.dateFormatterPattern());
+
+        ZonedDateTime lastPostedDate =
+                getLatestPostDateFromItems(rssItems, feedConfig.dateFormatterPattern());
+        updateLastDateToDatabase(feedConfig, rssFeedRecord, lastPostedDate);
+
+        if (lastSavedDate.isEmpty()) {
+            return;
+        }
+
+        final Predicate<Item> shouldItemBePosted = item -> {
+            ZonedDateTime itemPubDate =
+                    getDateTimeFromItem(item, feedConfig.dateFormatterPattern());
+            return itemPubDate.isAfter(lastSavedDate.get());
+        };
+
+        rssItems.reversed()
+            .stream()
+            .filter(shouldItemBePosted)
+            .forEach(item -> postItem(textChannel, item, feedConfig));
+    }
+
+    /**
+     * Retrieves an RSS feed record from the database based on the provided RSS feed configuration.
+     *
+     * @param feedConfig the RSS feed configuration to retrieve the record for
+     * @return the RSS feed record retrieved from the database
+     */
+    private RssFeedRecord getRssFeedRecordFromDatabase(RSSFeed feedConfig) {
+        return database.read(context -> context.selectFrom(RSS_FEED)
+            .where(RSS_FEED.URL.eq(feedConfig.url()))
+            .limit(1)
+            .fetchAny());
+    }
+
+    /**
+     * Retrieves the last saved date from the database record associated with the given RSS feed
+     * record.
+     * <p>
+     * If the RSS feed record is null, returns an empty {@link Optional}.
+     *
+     * @param rssRecord the RSS feed record to retrieve the last saved date from
+     * @param dateFormatterPattern the pattern used to parse the date from the database record
+     * @return An {@link Optional} containing the last saved date if it could be retrieved and
+     *         parsed successfully, otherwise an empty {@link Optional}
+     */
+    private Optional<ZonedDateTime> getLastSavedDateFromDatabaseRecord(RssFeedRecord rssRecord,
+            String dateFormatterPattern) {
+        if (rssRecord == null) {
+            return Optional.empty();
+        }
+
+        try {
+            ZonedDateTime savedDate =
+                    getZonedDateTime(rssRecord.getLastDate(), dateFormatterPattern);
+            return Optional.of(savedDate);
+        } catch (DateTimeParseException e) {
+            return Optional.empty();
+        }
+    }
+
+    /**
+     * Retrieves the latest post date from the given list of items.
+     *
+     * @param items the list of items to retrieve the latest post date from
+     * @param dateFormatterPattern the pattern used to parse the date from the database record
+     * @return the latest post date as a {@link ZonedDateTime} object, or null if the list is empty
+     */
+    private ZonedDateTime getLatestPostDateFromItems(List<Item> items,
```

Comment:
So, either always use optional, or always use nullable, it's confusing that you use both, even more confusing when it's all in the same class, it's really hard to predict what a method can return, thus comment apply to everywhere in this class.